From 14cea0c0ddca9ee451e0f9ce40edd97510b0e7b4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 29 Jan 2018 18:31:19 +0100
Subject: [PATCH 1/1] efi_loader: call efi_setup_loaded_image() first

In do_bootefi_exec() the first thing we should do is to initialize the
object list. Everything else depends on this.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 cmd/bootefi.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmd/bootefi.c b/cmd/bootefi.c
index e85bff96c0..fc88be809a 100644
--- a/cmd/bootefi.c
+++ b/cmd/bootefi.c
@@ -169,6 +169,10 @@ static efi_status_t do_bootefi_exec(void *efi, void *fdt,
 	const efi_guid_t fdt_guid = EFI_FDT_GUID;
 	bootm_headers_t img = { 0 };
 
+	/* Initialize and populate EFI object list */
+	if (!efi_obj_list_initalized)
+		efi_init_obj_list();
+
 	/*
 	 * Special case for efi payload not loaded from disk, such as
 	 * 'bootefi hello' or for example payload loaded directly into
@@ -183,10 +187,6 @@ static efi_status_t do_bootefi_exec(void *efi, void *fdt,
 		assert(device_path && image_path);
 	}
 
-	/* Initialize and populate EFI object list */
-	if (!efi_obj_list_initalized)
-		efi_init_obj_list();
-
 	ret = efi_create_handle(&handle);
 	if (ret != EFI_SUCCESS)
 		return EFI_OUT_OF_RESOURCES;
-- 
2.14.2

